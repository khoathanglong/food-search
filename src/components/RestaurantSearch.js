import { TextField, InputAdornment, Grid } from "@mui/material";
import { Search } from "@mui/icons-material";
import { useState } from "react";

import { Categories } from "./Categories";
import { Restaurants } from "./Restaurants";

export const RestaurantSearch = () => {
  const [currentCategory, setCurrentCategory] = useState("all");
  const [searchText, setSearchText] = useState("");

  const handleSearchTextChange = (e) => {
    setSearchText(e.target.value);
  };

  const handleSelectCategory = (category) => {
    setCurrentCategory(category);
  };

  return (
    <Grid container spacing={2}>
      <Grid item xs={12} md={4}>
        <TextField
          fullWidth={true}
          placeholder="Enter restaurant name..."
          variant="outlined"
          size="small"
          InputProps={{
            startAdornment: (
              <InputAdornment position="start">
                <Search />
              </InputAdornment>
            ),
            sx: { borderRadius: 3 },
          }}
          onChange={handleSearchTextChange}
        />
      </Grid>
      <Grid item xs={12}>
        <Categories
          onChange={handleSelectCategory}
          currentCategory={currentCategory}
        />
      </Grid>

      <Grid item xs={12}>
        <Restaurants
          currentCategory={currentCategory}
          searchText={searchText}
        />
      </Grid>
    </Grid>
  );
};
