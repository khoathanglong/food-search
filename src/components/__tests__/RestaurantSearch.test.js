import { RestaurantSearch } from "../RestaurantSearch";
import { fireEvent, render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";

jest.mock("../Categories", () => ({
  Categories: (props) => {
    const MockName = "mock-category";
    return <MockName role="mockcategory" {...props} />;
  },
}));
jest.mock("../Restaurants", () => ({
  Restaurants: (props) => {
    const MockName = "mock-restaurants";
    return <MockName role="mockrestaurants" {...props} />;
  },
}));
describe("RestaurantSearch", () => {
  it("renders correctly", () => {
    const { container } = render(<RestaurantSearch />);
    expect(container).toMatchSnapshot();
  });

  it("updates searchText correctly", () => {
    const { container } = render(<RestaurantSearch />);

    fireEvent.change(container.querySelector("input"), {
      target: { value: "hello world" },
    });
    expect(screen.getByDisplayValue("hello world")).toBeTruthy();
  });
});
