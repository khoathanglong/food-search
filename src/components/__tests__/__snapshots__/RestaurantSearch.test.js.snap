// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestaurantSearch renders correctly 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 css-1lj5egr-MuiGrid-root"
    >
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedStart css-1w3v0z0-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeSmall css-ittuaa-MuiInputAdornment-root"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="SearchIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
          </div>
          <input
            aria-invalid="false"
            class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedStart css-1ua80n0-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r0:"
            placeholder="Enter restaurant name..."
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-hdw1oc"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-eq2k9k-MuiGrid-root"
    >
      <mock-category
        currentcategory="all"
        role="mockcategory"
      />
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <mock-restaurants
        currentcategory="all"
        role="mockrestaurants"
        searchtext=""
      />
    </div>
  </div>
</div>
`;
